packages/perspective-jupyterlab/test/jupyter/utils.js

```diff
@@ -109,15 +112,23 @@ module.exports = {
             visible: true,
         });
 
-        // wait for a cell to be active
-        await page.waitForSelector(
-            '.jp-Notebook-ExecutionIndicator:not([data-status="idle"])',
-            {timeout: 3000}
-        );
+        // if theres a "select kernel" dialog, just click it
+        try {
+            const button = await page.waitForSelector(
+                "div.jp-Dialog-content > div.jp-Dialog-footer > button.jp-mod-accept",
+                {timeout: 5000}
+            );
+            if (button) {
+                button.click();
+            }
+        } catch {
+            // ignore
+        }
 
+        // wait for a cell to be active
         await page.waitForSelector(
             '.jp-Notebook-ExecutionIndicator[data-status="idle"]',
-            {timeout: 3000}
+            {timeout: 15000}
```

Comment:
Debug code (I hope ...)

Reply:
This is max wait time, too short and you get spurious failures, too long and you wait too long before real failures.